plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -657,29 +637,135 @@ describe('KubernetesProxy', () => {
         },
       ),
     );
-    permissionApi.authorize.mockResolvedValue([
-      { result: AuthorizeResult.ALLOW },
-    ]);
+
     clusterSupplier.getClusters.mockResolvedValue([
       {
         name: 'cluster1',
         url: 'http://localhost:9999/subpath',
         authProvider: '',
       },
     ]);
+
     authTranslator.decorateClusterDetailsWithAuth.mockImplementation(
       async x => x,
     );
-    const app = express().use(
-      Router().use('/mountpath', proxy.createRequestHandler({ permissionApi })),
-    );
 
-    const requestPromise = request(app)
-      .get('/mountpath/api/v1/namespaces')
-      .set(HEADER_KUBERNETES_CLUSTER, 'cluster1');
-    worker.use(rest.all(requestPromise.url, (req: any) => req.passthrough()));
+    const requestPromise = setupProxyPromise({
+      proxyPath: '/mountpath',
+      requestPath: '/api/v1/namespaces',
+
+      headers: {
+        [HEADER_KUBERNETES_CLUSTER]: 'cluster1',
+      },
+    });
+
     const response = await requestPromise;
 
     expect(response.status).toEqual(200);
   });
+
+  describe('WebSocket', () => {
+    const proxyPath = '/proxy';
+    const proxyPort = 9000;
+    const wsPath = '/ws';
+    const wsPort = 9999;
+
+    let wsServer: WebSocketServer;
+    let expressApp: express.Express;
+    let expressServer: Server;
+
+    const eventPromiseFactory = (
+      ws: WebSocket,
+      event: 'connection' | 'open' | 'close' | 'error' | 'message',
+    ) => new Promise(resolve => ws.once(event, x => resolve(x?.toString())));
+
+    beforeAll(async () => {
+      await new Promise(resolve => {
+        expressApp = express().use(
+          Router()
+            .use(proxyPath, proxy.createRequestHandler({ permissionApi }))
+            .use(errorHandler()),
+        );
+
+        expressServer = expressApp.listen(proxyPort, '0.0.0.0', () => {
+          resolve(null);
+        });
+      });
+
+      wsServer = new WebSocketServer({ port: wsPort, path: wsPath });
+
+      wsServer.on('connection', (ws: WebSocket) => {
+        // send immediatly a feedback to the incoming connection
```

Comment:
I'm have a low tolerance for comments so I'd say either nothing:
```suggestion
```
or just polish the English in the comment:
```suggestion
        // immediately send feedback to the incoming connection
```

Reply:
I think it's obvious what the send in the next line is doing, will remove.